### ISSUE TITLE
[COGNITION]: GIBCT EYB - Learn more modal triggers MUST be moved outside labels for VETTEC estimated benefits

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#14417]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a user, I expect label text to automatically focus the nearest input (should be related semantically) and open the Learn more modal if I click that string of text. Currently, the VETTEC estimate your benefits inputs do not set focus when I click on the non-button text.

## Details

I need to be able to click on the `<label>` text to set focus on the semantically related input, **and** click on the "Learn more" buttons to launch modal windows in the VETTEC estimate your benefits section. Screenshot attached below.

## Acceptance Criteria

- [ ] Clicking on the label > span text that's not the "Learn more" button should focus the related input
- [ ] Visual design should not change
- [ ] Screen readers should announce the labels properly and trigger the Learn more links properly
- [ ] All behaviors should be consistent with the OTJ and higher education profile views

## Definition of done

1. Review and acknowledge feedback.
1. Update code to match pattern on higher education and OTJ profile views.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Environment

* https://staging.va.gov/gi-bill-comparison-tool/profile/3V000142/CERTIFIED%20INFORMATION%20SYSTEMS%20SECURITY or any other VETTEC profile view

## Steps to Recreate

1. Enter https://staging.va.gov/gi-bill-comparison-tool/profile/3V000142/CERTIFIED%20INFORMATION%20SYSTEMS%20SECURITY in browser
1. Click on the "Tuition and fees for program" text string under Estimate your benefits
1. Verify the input does not receive focus (this is the error)

## WCAG or Vendor Guidance (optional)

* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

## Screenshots or Trace Logs

![Screen Shot 2020-10-05 at 1 59 51 PM](https://user-images.githubusercontent.com/934879/95129362-87957900-0720-11eb-916f-117c3f295a5c.png)

**Comment by hughes-dustin** [comment]
branch: bah-14417
PR: https://github.com/department-of-veterans-affairs/vets-website/pull/14513

**Comment by sonntag-adam** [comment]
@1Copenut this is in staging.

**Comment by 1Copenut** [comment]
@sonntag-adam I re-tested briefly with iOS + VoiceOver. The Learn more links weren't be announced like the non-VETTEC schools, so I looked a bit closer at the code snippet.

The VETTEC schools have a label wrapped inside another label, then the button with no span. This is different than the non-VETTEC schools' labels + learn more buttons.

Can we take another look at this and refactor to match the non-VETTEC code? I'll test again and anticipate it working as it should.

**Comment by hughes-dustin** [comment]
@1Copenut Looking into your feedback now. I'll see what I can do.

**Comment by ksantiagomorgan6** [comment]
Additional PR: https://github.com/department-of-veterans-affairs/vets-website/pull/15025https://github.com/department-of-veterans-affairs/vets-website/pull/15025

**Comment by delli-gatti-michael** [comment]
Local testing complete

**Comment by delli-gatti-michael** [comment]
@1Copenut This fix is now on Staging

**Comment by 1Copenut** [comment]
@delli-gatti-michael I spot checked in staging and things look good. Moving this to close.
